src/tests/Microsoft.PowerFx.Connectors.Tests/BasicRestTests.cs

```diff
@@ -50,6 +52,24 @@ public async Task BasicHttpCall()
             AssertLog(testConnector, "GET http://localhost:5000/Keys?keyName=Key1");
         }
 
+        [Fact]
+        public async void BasicHttpCallWithHeader()
+        {
+            var testConnector = new LoggingTestServer(@"Swagger\TestOpenAPI2.json");
+            testConnector.SetResponse(@"[{""date"":""2022-06-09T17:43:33.6483791+02:00"",""temperatureC"":-15,""temperatureF"":6,""summary"":""Bracing"",""index"":121},{""date"":""2022-06-10T17:43:33.6483939+02:00"",""temperatureC"":46,""temperatureF"":114,""summary"":""Freezing"",""index"":121},{""date"":""2022-06-11T17:43:33.6483941+02:00"",""temperatureC"":3,""temperatureF"":37,""summary"":""Bracing"",""index"":121},{""date"":""2022-06-12T17:43:33.6483943+02:00"",""temperatureC"":34,""temperatureF"":93,""summary"":""Warm"",""index"":121},{""date"":""2022-06-13T17:43:33.6483945+02:00"",""temperatureC"":27,""temperatureF"":80,""summary"":""Mild"",""index"":121}]");
+
+            var config = new PowerFxConfig();
+            config.AddService("Test", testConnector._apiDocument, new HttpClient(testConnector) { BaseAddress = _fakeBaseAddress });
+
+            var engine = new RecalcEngine(config);
+
+            var r1 = await engine.EvalAsync("Test.GetWeatherWithHeader({ id : 11 })", CancellationToken.None);
+            dynamic i1 = (ExpandoObject)((object[])r1.ToObject())[0];
```

Comment:
>(ExpandoObject)((object[])r1.ToObject())[0]; [](http://example.com/codeflow?start=25&length=44)

Can we avoid this? We don't want customers writing code like this...

or possibley:
dynamic i1 = r1.ToObject();
Assert.Equals(121, r1[0].index);

Or we could move this into the expression, like
"First(Test.Get(...)).index"

Reply:
:)